Mathlib/RingTheory/Polynomial/ScaleRoots.lean

```diff
@@ -227,10 +227,12 @@ lemma add_scaleRoots_of_natDegree_eq (p q : R[X]) (r : R) (h : natDegree p = nat
     r ^ (natDegree p - natDegree (p + q)) • (p + q).scaleRoots r =
       p.scaleRoots r + q.scaleRoots r := by
   ext n; simp only [coeff_smul, coeff_scaleRoots, coeff_add, smul_eq_mul,
-    mul_comm (r ^ _), mul_assoc, ← pow_add, ← h, ← add_mul, add_comm (_ - n)]
+    mul_comm (r ^ _), ← pow_add, ← h, ← add_mul, add_comm (_ - n)]
+  rw [mul_assoc, ← pow_add]
```

Comment:
Why do these not fire in simp now?

Reply:
They do fire, but `mul_assoc` induces a max rec depth error.

I've added an "adaptation note" explaining this, but am inclined to ship.